### ISSUE TITLE
rendering the name of a user associated with a model

### ISSUE THREAD

**Issue by BobWalsh** [TrestleAdmin/trestle#88]
```
Post 
belongs_to :user
User
has_many :posts
```

Out of the box, Trestle renders the association correctly, showing a dropdown of user names:
![image](https://user-images.githubusercontent.com/5097/36861932-3d8c13a0-1d3a-11e8-9bdf-93f75d154962.png)

But when I attempt to modify in posts_admin my form:
```
form do |post|
    select :user_id
    text_field :title
    check_box :published
    check_box :frontpage
    datetime_field :frontpage_datetime
    editor :body
    editor :checklist
  end
```
... I get just a dropdown of User ids.
![image](https://user-images.githubusercontent.com/5097/36861967-593a44c8-1d3a-11e8-9161-51ce61689de1.png)

Bottom line, I can't figure out Trestle's magic, just from looking at `https://github.com/TrestleAdmin/trestle/tree/3f6e81a724a2c3ef9e9def522db5adeca2919015/lib/trestle/form/fields`

Is there a better place to look and what should `select :id` be?

**Comment by svoboda-jan** [comment]
Did you mean `user_id`? Like `select :user_id, User.all`.

**Comment by BobWalsh** [comment]
Thanks, that worked! 

It would be helpful if there was documentation on this...

**Comment by svoboda-jan** [comment]
Yes, I agree there should be more and better documentation.

If that solves the problem, please close this issue.

**Comment by BobWalsh** [comment]
It did, but on another simple project:
Cohorts have many shortcuts, shortcuts belongs_to a Cohort,

With shortcuts_admin.rb:
```
form do |shortcuts|
    text_field :question
    select :cohort_id, Cohort.all
    text_field :answer
    text_field :citation_title
    text_field :citation_url
    check_box  :testscale
    text_field :keycaps
    text_field :description
    text_field :gif_url
    static_field :id
  end
```
Trestle returns:
![cursor_and_editing_shortcut____devshortcuts](https://user-images.githubusercontent.com/5097/38371841-4d6855a8-38a2-11e8-9f28-74ca686de7f8.png)

How does Trestle "pick" which field to display in the select?

**Comment by spohlenz** [comment]
Trestle tries each of the methods in order from `Trestle.config.display_methods` (which defaults to `[:display_name, :full_name, :name, :title, :username, :login, :email]`, plus `#to_s` is also tried as a last resort). If the instance `responds_to?` that method, then it calls it and returns the result.

If no matching methods are found, it will default to `"#{klass} (#{id})"`, as you are seeing.

**Comment by BobWalsh** [comment]
Thanks! - open to a PR that improves the docs for this project?
